### PR TITLE
Add Inventory Backend

### LINKED ISSUE #362: Inventory api
Write post request: 
1. Get all items(/getItems) *search by category
2. Edit item(/editItem) *search by name, token validation
3. Delete item(/deleteItem) *search by name, token validation
4. Add item(/addItem) *token validation

### PR DESCRIPTION
**Pull request by aashrav**
Created inventory model, routing, and api test cases

Resolves #362

### CODE REVIEW

File: test/InventoryItem.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+describe('InventoryItem', () => {
+  before(done => {
+    app = tools.initializeServer();
+    // Before each test we empty the database
+    tools.emptySchema(InventoryItem);
+    done();
+  });
+
+  // What does these do
```
Comment (evanugarte) [inline test/InventoryItem.js:33]: @ThaiQ can you explain what these do

File: test/InventoryItem.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    picture: '%&(&%##$^&(&#@'
+  };
+  const ITEM_WITH_INVALID_TOKEN = {
+    token: 'invalid'
+  };
+  const ITEM_WITHOUT_REQUIRED_FIELDS = {
+    name: 'chocolate'
+  };
+  const ITEM_WITH_INVALID_NAME = {
+    name: 'bob'
```
Comment (evanugarte) [inline test/InventoryItem.js:64]: hahaha

File: test/InventoryItem.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    category: 'Snack',
+    description: 'A snack that you can eat',
+    picture: ')(^#!@$%*(*'
+  };
+
+  describe('/POST addItem', () => {
+    it('Should return 403 when an invalid token is supplied', done => {
+      chai
+        .request(app)
+        // can you explain this routing
```
Comment (evanugarte) [inline test/InventoryItem.js:79]: @ThaiQ

**Comment by evanugarte** [comment]
@aashrav rebase!

File: test/InventoryItem.js (hunk 3 above)
Comment (aashrav) [reply test/InventoryItem.js:79]: Sorry have to take out those comments...Already asked thai those questions

**Pull request merged**